imap/ical_support.c

```diff
@@ -376,6 +376,77 @@ static int span_compare_range(icaltime_span *span, icaltime_span *range)
     return 0; /* span overlaps range */
 }
 
+struct multirrule_iterator_entry {
+    struct icaltimetype next;
+    icalrecur_iterator *icaliter;
+};
+
+struct multirrule_iterator {
```

Comment:
I think I did not use ptrarray_t so that we can cleanly port this upstream to libical. But in that case we could just as well use icalarray. Now that I think of it I might just as well refactor this to use a dynarray, which should avoid a few cache misses.